src/openrct2/scripting/bindings/entity/ScVehicle.cpp

```diff
@@ -187,42 +187,42 @@ namespace OpenRCT2::Scripting
         {
             if (value.type() == DukValue::Type::NUMBER)
             {
-                vehicle->next_vehicle_on_train = static_cast<uint16_t>(value.as_int());
+                vehicle->next_vehicle_on_train = EntityId::FromUnderlying(value.as_int());
             }
             else
             {
-                vehicle->next_vehicle_on_train = SPRITE_INDEX_NULL;
+                vehicle->next_vehicle_on_train = EntityId::GetNull();
             }
         }
     }
 
     uint16_t ScVehicle::previousCarOnRide_get() const
     {
         auto vehicle = GetVehicle();
-        return vehicle != nullptr ? vehicle->prev_vehicle_on_ride : 0;
+        return vehicle != nullptr ? vehicle->prev_vehicle_on_ride.ToUnderlying() : EntityId::GetNull().ToUnderlying();
     }
     void ScVehicle::previousCarOnRide_set(uint16_t value)
     {
         ThrowIfGameStateNotMutable();
         auto vehicle = GetVehicle();
         if (vehicle != nullptr)
         {
-            vehicle->prev_vehicle_on_ride = value;
+            vehicle->prev_vehicle_on_ride = EntityId::FromUnderlying(value);
         }
     }
 
     uint16_t ScVehicle::nextCarOnRide_get() const
     {
         auto vehicle = GetVehicle();
-        return vehicle != nullptr ? vehicle->next_vehicle_on_ride : 0;
+        return vehicle != nullptr ? vehicle->next_vehicle_on_ride.ToUnderlying() : EntityId::GetNull().ToUnderlying();
```

Comment:
This looks more correct than what was there previously, but I wonder if plug-ins rely on this (incorrect) behaviour.

Reply:
I'm not sure what do about it, entity 0 is valid and that would be problematic over time. There is also no constant in js for the magic null that I could just update.

Reply:
This was wrong anyway, because if `next_vehicle_on_ride` is SPRITE_INDEX_NULL, it should return null in JS.

Reply:
It uses now DukValue and I changed the ts docs.